File: ui/src/backend/payloads/admin/adminaddresspayload.ts
import { GlobalUserDatePayload } from "../global/GlobalUserDatePayload";

export class AdminAddressPayload  extends GlobalUserDatePayload  {
    address: string | null;
    city: string | null;
    town: string | null;
    country: string | null;
    postalCode: string | null;
    latitude: number | null;
    longitude: number | null;
    
    constructor(){
        super();
        this.address = null;
        this.city = null;
        this.town = null;
        this.country = null;
        this.postalCode = null;
        this.latitude = null;
        this.longitude = null;
    }
}

File: api/src/main/java/com/mystore/manager/api/admin/payload/AdminAddressPayload.java
package com.mystore.manager.api.admin.payload;

import com.mystore.manager.api.common.payload.GlobalUserDatePayload;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class AdminAddressPayload extends GlobalUserDatePayload {
    private String address;
    private String city;
    private String town;
    private String country;
    private String postalCode;
    private Double latitude;
    private Double longitude;
}
